adapters/lifestreet_test.go

```diff
@@ -0,0 +1,298 @@
+				Price: tag.bid,
+				AdM:   tag.content,
+				W:     lsdata.width,
+				H:     lsdata.height,
+			}
+		}
+	}
+	if bid == nil {
+		http.Error(w, fmt.Sprintf("Slot tag '%s' not found", breq.Imp[0].TagID), http.StatusInternalServerError)
+		return
+	}
+
+	resp := openrtb.BidResponse{
+		ID:    "2345676337",
+		BidID: "975537589956",
+		Cur:   "USD",
+		SeatBid: []openrtb.SeatBid{
+			{
+				Seat: "LSM",
+				Bid:  []openrtb.Bid{*bid},
+			},
+		},
+	}
+
+	if lsdata.delay > 0 {
+		<-time.After(lsdata.delay)
+	}
+
+	js, err := json.Marshal(resp)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	w.Header().Set("Content-Type", "application/json")
+	w.Write(js)
+}
+
+func TestLifestreetBasicResponse(t *testing.T) {
+	server := httptest.NewServer(http.HandlerFunc(DummyLifestreetServer))
+	defer server.Close()
+
+	lsdata = lsBidInfo{
+		appBundle:            "AppNexus.PrebidMobileDemo",
+		deviceIP:             "111.111.111.111",
+		deviceUA:             "Mozilla/5.0 (iPhone; CPU iPhone OS 10_3_1 like Mac OS X) AppleWebKit/603.1.30 (KHTML, like Gecko) Mobile/14E8301",
+		deviceMake:           "Apple",
+		deviceModel:          "x86_64",
+		deviceConnectiontype: 1,
+		deviceIfa:            "6F3EA622-C2EE-4449-A97A-AE986D080C08",
+		tags:                 make([]lsTagInfo, 2),
+		referrer:             "http://test.com",
+		width:                320,
+		height:               480,
+	}
+	lsdata.tags[0] = lsTagInfo{
+		code:    "first-tag",
+		slotTag: "slot123.123",
+		bid:     2.44,
+	}
+	lsdata.tags[1] = lsTagInfo{
+		code:    "second-tag",
+		slotTag: "slot122.122",
+		bid:     1.11,
+	}
+
+	conf := *DefaultHTTPAdapterConfig
+	an := NewLifestreetAdapter(&conf, server.URL)
+	an.URI = server.URL
+
+	pbin := pbs.PBSRequest{
+		AdUnits: make([]pbs.AdUnit, 2),
+		App: &openrtb.App{
+			Bundle: lsdata.appBundle,
+		},
+		Device: &openrtb.Device{
+			UA:             lsdata.deviceUA,
+			IP:             lsdata.deviceIP,
+			Make:           lsdata.deviceMake,
+			Model:          lsdata.deviceModel,
+			Connectiontype: lsdata.deviceConnectiontype,
+			IFA:            lsdata.deviceIfa,
+		},
+	}
+	for i, tag := range lsdata.tags {
+		pbin.AdUnits[i] = pbs.AdUnit{
+			Code: tag.code,
+			Sizes: []openrtb.Format{
+				{
+					W: lsdata.width,
+					H: lsdata.height,
+				},
+			},
+			Bids: []pbs.Bids{
+				pbs.Bids{
+					BidderCode: "lifestreet",
+					BidID:      fmt.Sprintf("random-id-from-pbjs-%d", i),
+					Params:     json.RawMessage(fmt.Sprintf("{\"slot_tag\": \"%s\"}", tag.slotTag)),
+				},
+			},
+		}
+	}
+
+	body := new(bytes.Buffer)
+	err := json.NewEncoder(body).Encode(pbin)
+	if err != nil {
+		t.Fatalf("Json encoding failed: %v", err)
+	}
+
+	fmt.Println("body", body)
+
+	req := httptest.NewRequest("POST", server.URL, body)
+	req.Header.Add("User-Agent", lsdata.deviceUA)
+	req.Header.Add("Referer", lsdata.referrer)
+	req.Header.Add("X-Real-IP", lsdata.deviceIP)
+
+	pc := pbs.ParseUIDCookie(req)
```

Comment:
these also need to be refactored to latest in master. (see here for example: https://github.com/prebid/prebid-server/blob/master/adapters/appnexus_test.go#L311)